### PR TITLE
Minor changes

### PR DESCRIPTION
**Pull request by mrthegreensky**
This pull request contains the following:
- [x] Ignoring of headerview when performing a long click.
- [x] Ignoring of footerview when performing of a long click.
- [x] Tags are able to be deleted successfully even when a claim is submitted.

### CODE REVIEW

File: Team9Project/src/com/indragie/cmput301as1/ExpenseClaimDetailActivity.java [hunk 1]
```diff
@@ -277,24 +280,31 @@ private void setEditable(){
...
-				new LongClickDeleteListener(this, new LongClickDeleteListener.OnDeleteListener() {
-					@Override
-					public void onDelete(int position) {
-						if (getListView().getItemAtPosition(position) == null) {
-							return;
-						}
-						longPressedItemPosition = itemPositionForListViewPosition(position);
-						ExpenseClaimDetailController.DetailItem.ItemType type = controller.getItemType(longPressedItemPosition);
-						startDeleteAlertDialog(type);
+		getListView().setOnItemLongClickListener(
```
Comment (indragiek) [inline Team9Project/src/com/indragie/cmput301as1/ExpenseClaimDetailActivity.java]: This should eventually be refactored to be part of LongClickDeleteListener but you can leave it this way for now if you want.

File: Team9Project/src/com/indragie/cmput301as1/LongClickDeleteListener.java [hunk 2]
```diff
@@ -65,33 +100,37 @@ public LongClickDeleteListener(Activity activity, OnDeleteListener onDeleteListe
...
 		longClickedPosition = position;
-		activity.startActionMode(new ActionMode.Callback() {
-			@Override
-			public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
-				switch (item.getItemId()) {
-				case R.id.action_delete:
-					onDeleteListener.onDelete(longClickedPosition);
-					mode.finish();
+		if(setStatus.set(position)) {
+			activity.startActionMode(new ActionMode.Callback() {
```
Comment (indragiek) [inline Team9Project/src/com/indragie/cmput301as1/LongClickDeleteListener.java:94]: Same here (LongClickDeleteListener)

**Comment by indragiek** [comment]
Travis says the build is broken, could you check the build log and see what's up?

File: Team9Project/src/com/indragie/cmput301as1/ExpenseClaimDetailActivity.java (hunk 1 above)
Comment (mrthegreensky) [inline Team9Project/src/com/indragie/cmput301as1/ExpenseClaimDetailActivity.java]: I did think about it, but then I thought that'd make them a bit too coupled? So I didn't proceed with it.

**Comment by mrthegreensky** [comment]
I have been looking it, but to be honest, it's kind of confusing me.

**Comment by indragiek** [comment]
It's still failing, here's the error:

```
 [javac] /home/travis/build/CMPUT301W15T09/Team9Project/Team9Project/src/com/indragie/cmput301as1/ExpenseClaimListActivity.java:80: error: constructor LongClickDeleteListener in class LongClickDeleteListener cannot be applied to given types;
    [javac]         getListView().setOnItemLongClickListener(new LongClickDeleteListener(this, new LongClickDeleteListener.OnDeleteListener() {
    [javac]                                                  ^
    [javac]   required: Activity,OnDeleteListener,SetStatus
    [javac]   found: ExpenseClaimListActivity,<anonymous OnDeleteListener>
    [javac]   reason: actual and formal argument lists differ in length
```

Did you make a change to `LongClickDeleteListener` that changed the constructor?

**Comment by mrthegreensky** [comment]
I did make a change to the constructor for LongClickDeleteListener but did change it when appropriately in the code for ExpenseClaimDetailActivity, which is what confuses me so much.

**Comment by indragiek** [comment]
The error is coming from `ExpenseClaimListActivity` (not detail), which you probably forgot to change it in :)

**Comment by mrthegreensky** [comment]
I thought that too, but I don't see it at all in the ExpenseClaimListActivity. 

The only thing I found to be close would be 

```
    getListView().setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
        @Override
        public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
            longPressedItemIndex = position;
            startActionMode(longClickCallback);
            return true;
        }
    });
```

**Comment by indragiek** [comment]
Try merging master, there was a change to that code in master.

**Comment by mrthegreensky** [comment]
Oh, okay. Will do! Thanks Indragie :)

**Pull request merged**